src/index.ts

```diff
@@ -1,5 +1,12 @@
 import { StateMachineProvider } from './StateMachineContext';
 import { createStore, useStateMachine } from './stateMachine';
 import { GlobalState } from './types';
+import { PersistOption } from './constants';
```

Comment:
maybe we can skip exporting this?

Reply:
Sure, I will slightly tweak it to meet all your requirements and will push the changes soon